jupyter_server/services/kernelspecs/handlers.py

```diff
@@ -48,6 +48,19 @@ def is_kernelspec_model(spec_dict):
     )
 
 
+def update_kernelspec_model(spec_dict):
+    """Returns the original kernelspec unless template substitutions are available."""
+    model = spec_dict
+    if "resource_dir" in spec_dict:
+        spec_str = json.dumps(spec_dict["spec"])
+        template_found = "{resource_dir}" in spec_str
+        resource_dir_exists = os.path.exists(spec_dict["resource_dir"])
+        if template_found and resource_dir_exists:
+            spec_str = spec_str.replace("{resource_dir}", spec_dict["resource_dir"])
```

Comment:
This change will result in the substitution of `{resource_dir}` with the **local** value (see the previous comment about non-local specs returned from a Gateway) which typically occurs at the [time of launch in `jupyter_client`](https://github.com/jupyter/jupyter_client/blob/c02471eece06d8725a6cd4d42940c2845795925a/jupyter_client/manager.py#L287).   

(I don't see any issue with that but just wanted to make that comment in case others do.)